source/fuzz/fuzzer_pass_add_loads.cpp

```diff
@@ -79,13 +83,36 @@ void FuzzerPassAddLoads::Apply() {
           return;
         }
 
+        auto choosen_instruction =
+            relevant_instructions[GetFuzzerContext()->RandomIndex(
+                relevant_instructions)];
+
+        bool is_atomic_load = false;
+        uint32_t memory_scope_id = 0;
+        uint32_t memory_semtantics_id = 0;
+
+        // has work group or storage buffer (storage class)
+        if (choosen_instruction) {
+          if (GetFuzzerContext()->ChoosePercentage(
+                  GetFuzzerContext()->GetChanceOfAddingAtomicLoad())) {
+            is_atomic_load = true;
+            memory_scope_id =
+                FindOrCreateConstant({SpvScopeInvocation},
+                                     FindOrCreateIntegerType(32, false), false);
+            memory_semtantics_id =
+                FindOrCreateConstant(/*NOTE: I think the value(Words) here needs
+                                        to be selected under a condition? */
+                                     {SpvMemorySemanticsWorkgroupMemoryMask,
+                                      SpvMemorySemanticsUniformMemoryMask},
+                                     FindOrCreateIntegerType(32, false), false);
+          }
+        }
+
         // Choose a pointer at random, and create and apply a loading
         // transformation based on it.
```

Comment:
This comment needs to be updated, as it is incorrect after your changes.

Reply:
Done.